CMakeLists.txt

```diff
@@ -4,6 +4,15 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
   message(FATAL_ERROR "Prevented in-tree build. Please create a build directory outside of the SDL source code and run \"cmake -S ${CMAKE_SOURCE_DIR} -B .\" from there")
 endif()
 
+# This needs to be set before project() calls
+if(CMAKE_SYSTEM_NAME MATCHES "iOS" OR CMAKE_SYSTEM_NAME MATCHES "tvOS")
```

Comment:
If you move these under the call to `SDL_DetectCMakePlatform()`, the cmake script defines `TVOS`, `IOS`, `VISIONOS` and `MACOS` CMake variables.